subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -327,6 +330,7 @@ static bool schedule_send(struct bt_mesh_ext_adv *adv)
 
 	if (atomic_test_and_clear_bit(adv->flags, ADV_FLAG_PROXY)) {
 		(void)bt_le_ext_adv_stop(adv->instance);
+		atomic_clear_bit(adv->flags, ADV_FLAG_PROXY_START);
```

Comment:
Would it make sense to clear `ADV_FLAG_PROXY_START` before calling `bt_le_ext_adv_stop`? E.g. if connection happens during scheduling that is caused by `bt_le_ext_adv_stop` (not sure if this possible though)? This would prevent from calling `schedule_send` from `connected` callback.